Fix shape bounds and transforms for VoxelCylinderShape

jjhembd opened the pull request:
<!--
Thanks for the Pull Request!

Please review [Contribution Guide](https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md) before opening your first Pull Request.

To ensure your Pull Request is reviewed and accepted quickly, please refer to our [Pull Request Guidelines](https://github.com/CesiumGS/cesium/blob/main/CONTRIBUTING.md#pull-request-guidelines).

-->

# Description

Recent updates to the 3D Tiles extension [3DTiles_bounding_volume_cylinder](https://github.com/CesiumGS/3d-tiles/tree/voxels/extensions/3DTILES_bounding_volume_cylinder) included a change in the way the shape bounds were defined. Instead of an oriented bounding box, the extension now defines the radius, angle, and height bounds, along with an optional transform specified as a rotation and translation.

This PR updates the test datasets to follow the new spec, and reworks the `getCylinderShape` helper in `Cesium3DTilesVoxelProvider` to read the new values.

Other changes include:
- Cleaned up references to cylinder bounds to consistently assume the `(radius, angle, height)` coordinate order everywhere. This fixes the naming and functionality of the `Bounds` and `Clipping` editors in `VoxelInspector`.
- Handled cylinder voxels in the `updateVerticalExaggeration` helper in `VoxelPrimitive`. This enables vertical exaggeration of cylinders, and also passes dynamic updates to the `modelMatrix` through to `primitive._exaggeratedModelMatrix` so that they are accounted for in the rendering.

## Issue number and link

Fixes https://github.com/CesiumGS/cesium/issues/12275.
Fixes https://github.com/CesiumGS/cesium/issues/12276.
Fixes https://github.com/CesiumGS/cesium/issues/12521.

## Testing plan

1. Run `VoxelPrimitiveSpec` and `Cesium3DTilesVoxelProviderSpec` locally.
2. Load the [testing Sandcastle from #12275](https://sandcastle.cesium.com/#c=lVdtb9s2EP4rRD7JsUs7SbsNcRqsdRssQNIGtVesTYKVls4WUYr0SMqRPfi/70i9mLKTtAMMSzw+9348nmIljSVLDg+gyWsi4YGMwPA8o589Lbo7iP16pKRlXIK+O+iRf+8kIXOhpnBKZkwY6DnClBm4YivQNzz+DjrcmoOKVdKmMckzZrmSIdHyDASq2dI2neGdvJOliRQKCzKJKhtL4mdVgLiUZgGxVfqaF1w6norDxCCBJjDN5+NUPVxoloG5AT2GWMkEfbY6B69ilsvY2UNutIohyTUTYy7nAiZcgFeCG0uObkQmZQvolHGwKTfUE1CYfw4bcsblW5XLxOBWHVcnaOxgk9UC6BzsdQ2qxNJYKAlRJxDDip8QU4OeFJNgcKVBD0071SOmLb4xeRL91iPuFzBJFy/E32IlKKEw//fbTYu6/WYl6RosS5hl3qbP70cvA2yssgUaJO3kCaZRCKAXVx/fTE6OnYCNy07sK9XEmtk4HTlLdpxwpKgslh8mkC60sspZTzX8k4Ox79AAFNjUQKQW7mmqJPNZQ6EWpV3BEgQ5f02OKgAhGmyuJcEMwAwrOPGee9MJKY1vxX8nI8MtbOksHWEqLcK2CFqQw3C5ai/XgQTnWK3Cp+h2cB9sxymTcqvisdxhPTXp8LjIyekEQsqMbdNwIRSzJ8dvtGarKHDhsKWuzE8tQzOZqGwMkLQ93XGtIN3W0ltRW81sSg3YT17UhzybuqMNSbSVHVqd5hB47HgltpSQOapNnKFrkQBL1sgxGOLjrB1vsu52m+w36FWJXrXRK6QE6NocjsVSfP2CLOtnk+sisEsa1qIa1UWpumirLpDSUl0rF6AXoP9CpoL0SdRS9wILe/g4g7N2tcOweo7hq3evzbCuGXZZUgS7JHUb8w7JgL5CeK3drU+a/a9+fbyn2h2xIzrY4dvHLV3MUH6XHCP60AWyFPvCkTtPcMVV4Ye9h860yv4wIkp7xPTIsuf091od6zGHfQmgqKiuhS4pOjunJtDfHLvbkrFLBvfI7Q3CXpY8Cz3aQucaQD4LPt6CpyKHZ7En9z6Mv75qUJvyZbPXBPEfr3C7c5ONSqqPYd2IylZy22i7Lwus6rPYyjNuXP82SiwhquQ6zKZ9nccamIUbzTNuOSIX1SXQI3FuLPaIlPlVhhOKwJ6geVGdltYIsagFGNSZqSW8ESJq97NlecdUOHTxKQEsSaIyQOHE1eKOmvNa23vavPWaLhI4cNp2p4YEXp2Giwqw6fiX2o/K4GVrpPJz1rVjpfsetgjDQEaMg4NmdCZWE+UHE7yKx4sUNERtJjpt7fbqTuUu8HI+HOBB8oVUm1nVwJ7yYFIIg/HYuBBsl6GeaTbH/mRL2gSvhVPybal40uxc4wAcXVSLS7nILZkZ/+zhMVa4jNfZ33WMQXMmSFa9dJyKJqU1FQeA2Sw3LpCVJJpV1U+rIz2fbhu9u2JxXmWYGKZBxqumyT3BzYKG0++TiSIZBismgs9TN0wx6+yQ8x5xRkBRXvjO7gRittpRnOJcL8GYwFqfAYom4Tj0jhvLZAzYvI5+oYPhnrNMLFLWmLxQD1HoS2+roDzom2/b8b/MaVC+rVsc1y996/hTcrwNs3HMBETY0EPm+vBUdfDjIT+YMnam7dGXq8sP795/Qvuchlr+9lT8XMvxZdnuO97eMY4jMTNWgGsUE6XElOm3ubVK4vfYNfsOZMrn8/JbrG5zUdWzGjPo/4vWyzJaPt4HvYMzY1cCzusc/s5xGNSW5FpElPYtZAuBHpr+NMdvPUvjOmmEnPVD1rOELwlPXj/yHUliwYzBnVkuMAlruDs4P+sjfo8Vy8/1h49L0AKrEmHp0flVSaSUnvVx+TinLYO3I/k/) and verify that the voxel expands as expected.
3. Test the "Voxel Inspector" => "Transform", "Bounds", and "Clipping" editors in the [Voxels]() and [Voxels in 3D Tiles](http://localhost:8080/Apps/Sandcastle/index.html?src=Voxels%20in%203D%20Tiles.html) Sandcastles, focusing on the cylindrical examples.

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have updated the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

ggetz commented:
Thanks @jjhembd! Some small comments from me on the code. 

The testing sandcastle now works as expected for me. 👍 

However, I noticed some strange behavior when adjusting the max angle in the inspector (this is again the provided testing Sandcastle, I just didn't hit the "Make bigger" button first). There appears to be some clipping on the right side of the frame:

<img width="843" alt="image" src="https://github.com/user-attachments/assets/23b3c116-c30b-4911-879e-5dc85a1b5d2a" />

I'm also seeing some unexpected behavior when adjusting the transforms in the inspector in both the "Voxels" and "Voxels in 3D Tiles" sandcastles. The translation doesn't seem to have any effect, and any scale adjustments end up flattening the shape completely:

<img width="683" alt="image" src="https://github.com/user-attachments/assets/975c392c-8008-4a76-b245-42fd6d87117d" />

jjhembd commented on packages/engine/Source/Scene/Cesium3DTilesVoxelProvider.js:
```diff
@@ -534,16 +535,39 @@ function getBoxShape(box, tileTransform) {
 }
 
 function getCylinderShape(cylinder, tileTransform) {
```
Correct, since we updated the sample data at the same time, the existing specs capture the behavior of the new data format.

I added a unit test for vertical exaggeration of cylinder voxels, which didn't work before this PR.

jjhembd commented:
Thanks @ggetz! I addressed your code comments. For your last two observations:
1. Strange behavior when adjusting the max angle: I had a little trouble reproducing it, but it sounds like another manifestation of an existing issue: https://github.com/CesiumGS/cesium/issues/11004. Does that issue capture what you're seeing?
2. Unexpected behavior of the transforms in Voxel Inspector: This is caused by the Voxel Inspector overwriting the model matrix, after we already used the model matrix to scale the voxel to the size of the Earth. See https://github.com/CesiumGS/cesium/issues/12276#issuecomment-2725955430. I fixed the Sandcastles by moving the scaling for the dummy datasets to the `VoxelProvider.prototype.globalTransform`, which is where it would normally appear in real-world datasets. Let me know if this resolves it on your end.

ggetz commented:
All looks fantastic. Thanks @jjhembd!
